doc/member_types.md

```diff
@@ -543,54 +545,50 @@ Patch<SomeObject>():
 ```
 
 
-
 ## Type Modifiers
 
-### `optional`
 
-A member with type modifier `optional` can have the placeholder value
-`None` assigned. `None` indicates that the member is initialized, but
-has no regular value. `None` can only be assigned and not used as a
-second operand in other operations. Additionally, operations other than
-assignment have no effect on members that have `None` assigned.
+### `abstract`
 
-Members of any data type can have the `optional` modifier. `optional`
-members still need to be initialized with either `None` or a regular
-value to not be abstract.
+`abstract` is a type modifier for the `object` data type. It signifies
+that references to abstract nyan objects can be assigned as member values.
 
 ```python
-SomeObject():
-    a : optional(int)                 # declaration of an int as optional
-    b : optional(float)               # declaration of a float as optional
-    c : optional(set(OtherObject))    # declaration of a set(OtherObject) as optional
-    d : optional(OtherObject)         # declaration of a nyan object reference to OtherObject as optional
-
 OtherObject():
-    a : optional(int) = None                 # declaration and initialization with None
-    b : optional(OtherObject) = OtherObject  # declaration and initialization with regular value
+    # is abstract because member 'x' is abstract
+    x : int
+
+ChildObject(OtherObject):
+    # not abstract because member 'x' now has a value
+    x = 5
+
+SomeObject():
+    a : abstract(OtherObject)                # declaration of a to allow abstract objects with type OtherObject
+    b : abstract(OtherObject) = OtherObject  # declaration of b to allow abstract objects with type OtherObject
+                                             # and initialization with ChildObject
+    c : OtherObject = ChildObject            # ALLOWED (ChildObject is non-abstract)
+    c : OtherObject = OtherObject            # NOT ALLOWED (OtherObject is abstract)
 ```
 
 #### Usage Examples
 
 
 ```python
 OtherObject():
-    pass
+    x : int
 
 ChildObject(OtherObject):
-    pass
+    x = 5
 
 SomeObject():
-    a : optional(int) = 5
-    b : optional(float) = None
-    c : optional(set(OtherObject)) = None
-    d : optional(OtherObject) = OtherObject
+    a : abstract(OtherObject) = OtherObject
+    b : abstract(OtherObject) = ChildObject
+    b : abstract(OtherObject) = ChildObject
 
 Patch<SomeObject>():
-    a = None         # result: a = None        (reassignment)
-    b += 10.0        # result: b = None        (no effect)
-    c = {}           # result: c = {}          (reassignment)
-    d = ChildObject  # result: d = ChildObject (reassignment)
+    a = ChildObject  # result: a = ChildObject (reassignment)
+    b = ChildObject  # result: b = ChildObject (no effect)
+    c = OtherObject  # result: c = OtherObject (reassignment)
 ```
```

Comment:
not sure if this example is so helpful, it's nearly the same as the explaination above

Reply:
you're right I removed it